### PR TITLE
Allow ~ in Version and Release?

### PR DESCRIPTION
**Pull request by sghill**
Is it possible redline is being a bit too aggressive in validating version and release input?

Building a package with a `~` in the version or release causes redline to throw an exception. It looks like `-` is the only prohibited character in rpm versions and releases from [the docs at rpm.org](http://rpm.org/max-rpm-snapshot/ch-rpm-file-format.html):

> The only restriction placed on the version is that it cannot contain a dash "-".

> The release can be thought of as the package's version. [...] Like the version number, the only restriction is that dashes are not allowed.

If I build a package `nameRequired-1.0~beta-1.noarch.rpm`, running `rpm -qip` seems to correctly parse the version:

```
Name        : nameRequired                 Relocations: (not relocatable)
Version     : 1.0~beta                          Vendor: 
Release     : 1                             Build Date: Wed Dec 21 19:39:19 2016
Install Date: (not installed)               Build Host: sghill-mbp
Group       : groupRequired                 Source RPM: (none)
Size        : 0                                License: 
Signature   : (none)
Packager    : sghill
URL         : 
Summary     : 
Architecture: noarch
Description :
```

Context:
* #18 stated `-`, `/`, and `~` are illegal characters for rpm
* #39 implemented the feature
* https://github.com/nebula-plugins/gradle-ospackage-plugin/issues/213 ran into the tilde issue
* a commenter on https://github.com/nebula-plugins/gradle-ospackage-plugin/issues/49 ran into the tilde issue

### CODE REVIEW

**Comment by sghill** [comment]
I did a bit more research on this issue. There is an ongoing discussion [here](https://fedorahosted.org/fpc/ticket/398). There is also a [draft release](https://fedoraproject.org/wiki/PackagingDrafts/TildeVersioning) of the _Guidelines for Versioning Fedora Packages_ that outlines the meaning of using `~` in a version.

It looks like this was formerly a problem with Fedora, but that is no longer the case:

> there's no currently supported Fedora release where rpm
wouldn't support it.

from December 21, 2013 [here](https://lists.fedoraproject.org/pipermail/packaging/2013-December/009899.html)

**Comment by devalru** [comment]
Since rpm 4.10 the tilde char in "version" and "release" is allowed. See 
http://rpm.org/wiki/Releases/4.10.0

File: src/test/java/org/redline_rpm/ant/RedlineTaskTest.java [hunk 1]
```diff
@@ -90,15 +90,6 @@ public void testBadVersion() throws Exception {
 			// Pass
 		}
 
-		// test version with illegal char ~
-		task.setVersion("1.0~beta");
-		try {
```
Comment (devalru) [inline src/test/java/org/redline_rpm/ant/RedlineTaskTest.java]: Why did you remove the rest ?

Comment (sghill) [inline src/test/java/org/redline_rpm/ant/RedlineTaskTest.java]: If we allow tilde, the current test fails. We could either change the test to assert the version is set successfully, or remove it. Since there would no longer be special handling for `~` with this PR, I opted to remove it -- but happy to make any changes.

**Comment by craigwblake** [comment]
I think this is fine as is.  Thanks for the research and the patch!

**Comment by sghill** [comment]
Thank you! When is the next release of redline expected?

**Comment by craigwblake** [comment]
Hi Steve, it's hard for me to find time to work on Redline these days, but
I'm going to do my best to get a release out by early next week.

Sorry for the delay.

Cheers,
Craig
